accepted/future-releases/constructor-tearoffs/faq.md

```diff
@@ -0,0 +1,262 @@
+
+**A:** Most likely not. There are no plans to change the `dart:mirrors` library, and it's just different syntax for the same declaration. The constructor name is still going to be just the class name, and that's what `dart:mirrors` expose.
+
+## Function value instantiation
+
+Until now you could only instantiate *tear-offs* of function declarations or instance methods. You could write:
+
+```dart
+T id<T>(T value) => value;
+int Function(int) intId = id; // implicitly instantiated with <int>.
+```
+
+but you couldn't instantiate function *values* with a type argument:
+
+```dart
+T Function<T>(T) id = <T>(T value) => value;
+int Function(int) intId = id; // INVALID
+```
+
+There were reasons for this, mainly worries about implementations not being able to be efficient. The implementors have told us that it's not a problem, so we now remove that restriction and allow you to instantiate any function-typed expression. The `INVALID` above becomes valid and well-typed.
+
+_This is not actually part of the constructor tear-off feature, it's considered a bug-fix because some (but not all) compilers already supported the feature, and it will be available in older language versions too. See language issue [#1812](https://github.com/dart-lang/language/pull/1812) for details. This feature-fix will still very likely ship at the same time as constructor tear-offs, and it's affecting the explicit instantiation feature, so we include it here._
+
+This also applies to *callable objects* (objects that have an interface type with a `call` method), which we treat like function values in most places.
+
+**Q:** Where does that even matter?
+
+**A:** If you have a generic function *value*, it's usually something you've received as a parameter at some point (if you knew which value it was, you'd just refer directly to a function declaration). It's probably going to be fairly rare to then need to instantiate that function to a specific type, instead of keeping it generic until it's called. It can happen. It makes explicit instantiation easier to explain too!
+
+Hypothetical example:
+
+```dart
+// Not a clue, mate. You tell me.
+```
+
+**Q:** Couldn't I just instantiate the `call` method using method instantiation anyway?
+
+**A:** Yes and no. You could for callable objects, and we'd even add the `.call` implicitly for you (we do that for any callable object in a function-typed context, before checking whether the types actually work). It just didn't work for *real* function values. Dart2js never implemented instantiated tear-off of the `call` method of function values because that would be equivalent to instantiating the function value itself, which wasn't a supported feature (until now), so your code would just crash. We recognized this and initially planned to disallow doing a instantiated tear-off of a function's `call` method. Instead it turned out we can just support it consistently.
```

Comment:
Maybe we can skip the history part, and just say how it works when the callable objects generalization has been achieved? (And then, briefly: "This didn't work previously"; most developers don't care exactly how or why it didn't work ;-)